zeppelin-web-angular/src/app/pages/workspace/notebook/action-bar/action-bar.component.less

```
@@ -113,3 +113,22 @@
     }
   }
 });
+
+.search-code {
+  padding: 0;
+
+  &-row {
+    display: flex;
+
+    .ant-input-group,
+    .ant-input {
+      height: 32px;
+    }
+
+    .search-row-btn-group {
+    display: flex;
+    align-items: center;
+    flex-wrap: nowrap;
+    }
+  }
+}
```

Could you try this code? It applies `border-radius` correctly to the appropriate elements.
```suggestion
::ng-deep .search-code {
  padding: 0;

  &-row {
    display: flex;

    .ant-input-group,
    .ant-input-group-addon,
    .ant-input {
      height: 32px;
      border-radius: 0;
    }

    .search-row-btn-group {
      display: flex;
      align-items: center;
      flex-wrap: nowrap;
    }

    .ant-btn-group > .ant-btn:first-child,
    .ant-btn-group > .ant-btn:last-child {
      border-radius: 0;
    }

    &:first-child {
      .ant-input-group-addon {
        border-top-left-radius: 4px;
      }
      .ant-btn:last-child {
        border-top-right-radius: 4px;
      }
    }

    &:last-child {
      .ant-input-group-addon {
        border-bottom-left-radius: 4px;
      }
      .ant-btn:last-child {
        border-bottom-right-radius: 4px;
      }
    }
  }
}

```

---

The code I suggested still doesn't fully solve the border conflicts between elements. I tried to adjust it, but Ant component style make it hard to fix cleanly.
@dididy Do you have any suggestions for handling this in a more maintainable way?

---

Thanks, it seems to work well. I've added those to the suggestion comment I wrote.

---

Thank you. It looks much better!!